src/oatpp/core/async/Processor.hpp

```diff
@@ -123,10 +138,13 @@ class Processor {
 
 public:
 
-  Processor()
-    : m_running(true)
-    , m_tasksCounter(0)
-  {}
+  Processor() = default;
+
+  ~Processor() {
```

Comment:
Let's move this logic to `Processor::stop()` method

Reply:
Keep in mind: If we move this logic to `Processor::stop()` and call the destructor without calling the `Processor::stop()` we get a call to std::terminate.

How about joining the thread in stop, but as a fallback also join it in the Destructor if the thread is still joinable?

Reply:
Currently `stop()` is always called before destruction.
Processor is used via `Executor::SubmissionProcessor` wrapper - which is possible another thing to refactor (but it's out of the scope of this PR).

So it's better to have all the joins in the `stop()` method - to follow the existing pattern.

> Keep in mind: If we move this logic to Processor::stop() and call the destructor without calling the Processor::stop() we get a call to std::terminate.

I'm ok with that. In this case `std::terminate` will indicate the API misuse.